examples/aws-iam-authenticator-chart.dhall

```diff
@@ -0,0 +1,104 @@
+let kubernetes = ../schemas.dhall
+
+let release = "wintering-rodent"
+
+let name = "aws-iam-authenticator"
+
+let fullName = "${release}-${name}"
+
+let version = "0.1.1"
+
+let chart = "${name}-${version}"
+
+let heritage = "dhall"
+
+in  kubernetes.DaemonSet::{
+    , metadata = kubernetes.ObjectMeta::{
+      , name = "AWS IAM authenticator metadata"
+      , labels = toMap
+        { app = name
+        , chart = chart
+        , release = release
+        , heritage = heritage
+        }
+      }
+    , spec = Some kubernetes.DaemonSetSpec::{
+      , updateStrategy = Some kubernetes.DaemonSetUpdateStrategy::{
+        , type = Some "RollingUpdate"
+        }
+      , template = kubernetes.PodTemplateSpec::{
+        , metadata = kubernetes.ObjectMeta::{
+          , name = name
```

Comment:
This is one of the places where I think you can ommit the `metadata.name` attribute
This is a manifestation of https://github.com/dhall-lang/dhall-kubernetes/issues/8#issuecomment-539775177 I think the ground rule is that only toplevel objects need `name`, any template objects can (and must?) omit it instead.

I think if you explicitly set `metadata.name` here, really weird things might happen.  Every pod will have the same name, e.g. there will never be more than 1 pod created eventhough the daemonset wants to create more than 1.  

Also I think you _must_ set `template.selector` otherwise the `pods` that the `daemonset` will create will not associate with the `daemonset` (See https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/)
It used to be the case that if you left out `selector` that it would be implicitly created, but I think that behaviour was removed recently. At least for `Deployment`; maybe not for `daemonset`.


 I'm poking @akshaymankar here just to be sure.

Reply:
This is a bug that we should fix.  `name` should not be a required attribute of `ObjectMeta`

Reply:
Yeah, I only added the `name` field because the type required it

For reference, this is based off of: https://github.com/helm/charts/blob/550096fcda27d7637a7a066240c61a4c6cb61f21/stable/aws-iam-authenticator/templates/daemonset.yaml

Reply:
Ah yeh that uses an older version of the `DaemonSet` API.  Ours requires the `selector`: https://github.com/dhall-lang/dhall-kubernetes/blob/master/types/io.k8s.api.apps.v1.DaemonSetSpec.dhall  so your example will hopefully give you a nice type error! (Woohoo dhall types!)

Reply:
@arianvp: The example type-checks because the default record for `DaemonSetSpec` defines an empty `selector` field

Reply:
Ahh yes. I remember now. We have a bug open for this as well :) https://github.com/dhall-lang/dhall-kubernetes/issues/78

Reply:
> I think if you explicitly set metadata.name here, really weird things might happen.

I checked with a daemonset, nothing weird happened, the controller just ignored the `name` field while creating pods. But it might confuse somebody reading a config.

> Also I think you must set `template.selector`

This is true, if you don't set it, it is a validation error, but only for daemonsets. Not sure if this _should_ be a type error given the OpenAPI spec doesn't say anything about it.